flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -1149,22 +1168,28 @@ public void initializeState(FunctionInitializationContext context) throws Except
             migrateNextTransactionalIdHindState(context);
         }
 
-        String taskName = getRuntimeContext().getTaskName();
-        // Kafka transactional IDs are limited in length to be less than the max value of a short,
-        // so we truncate here if necessary to a more reasonable length string.
-        if (taskName.length() > maxTaskNameSize) {
-            taskName = taskName.substring(0, maxTaskNameSize);
-            LOG.warn(
-                    "Truncated task name for Kafka TransactionalId from {} to {}.",
-                    getRuntimeContext().getTaskName(),
-                    taskName);
+        String transactionalIdPrefix;
```

Comment:
nit: rename `transactionalIdPrefix` -> `actualTransactionalIdPrefix` or `computedTransactionalIdPrefix`?

Reply:
Great idea! It makes the variable more clear to be different from the `this.transactionalIdPrefix`.